neuraxle/hyperparams/space.py

```diff
@@ -167,13 +157,73 @@ def to_nested_dict_as_ordered_dict_primitive(self) -> OrderedDict:
         return flat_to_nested_dict(self, dict_ctor=OrderedDict)
 
 
-class HyperparameterSpace(HyperparameterSamples):
-    """Wraps an hyperparameter nested dict or flat dict, and offer a few more functions to process
+def nested_dict_to_flat(nested_dict, dict_ctor=OrderedDict):
+    """
+    Convert a nested hyperparameter dictionary to a flat one.
+
+    :param nested_dict: a nested hyperparameter dictionary.
+    :param dict_ctor: ``OrderedDict`` by default. Will use this as a class to create the new returned dict.
+    :return: a flat hyperparameter dictionary.
+    """
+    ret = dict_ctor()
+    for k, v in nested_dict.items():
+        if isinstance(v, dict) or isinstance(v, OrderedDict) or isinstance(v, dict_ctor):
+            _ret = nested_dict_to_flat(v, dict_ctor)
+            for key, val in _ret.items():
+                ret[k + PARAMS_SPLIT_SEQ + key] = val
+        else:
+            ret[k] = v
+    return ret
+
+
+def flat_to_nested_dict(flat_dict, dict_ctor=dict):
+    """
+    Convert a flat hyperparameter dictionary to a nested one.
+
+    :param flat_dict: a flat hyperparameter dictionary.
+    :param dict_ctor: ``OrderedDict`` by default. Will use this as a class to create the new returned dict.
+    :return: a nested hyperparameter dictionary.
+    """
+    pre_ret = dict_ctor()
+    ret = dict_ctor()
+    for k, v in flat_dict.items():
+        k, _, key = k.partition(PARAMS_SPLIT_SEQ)
+        if len(key) > 0:
+            if k not in pre_ret.keys():
+                pre_ret[k] = dict_ctor()
+            pre_ret[k][key] = v
+        else:
+            ret[k] = v
+    for k, v in pre_ret.items():
+        ret[k] = flat_to_nested_dict(v, dict_ctor)
+    return ret
+
+
+class HyperparameterSamples(RecursiveDict):
+    """
+    Wraps an hyperparameter nested dict or flat dict, and offer a few more functions.
+
+    This can be set on a Pipeline with the method ``set_hyperparams``.
+
+    HyperparameterSamples are often the result of calling ``.rvs()`` on an HyperparameterSpace.
+    """
+
+    def __init__(self, *args, **kwds):
+        super().__init__(RecursiveDict.DEFAULT_SEPARATOR, *args, **kwds)
```

Comment:
The separator should be provided in second argument. 
```suggestion
    def __init__(self, *args, separator=CONSTANT, **kwds):
        super().__init__(*args, separator=CONSTANT, **kwds)
```